### DISCUSSION TITLE
github desktop on mac fatal error with 'Icon?'

### DISCUSSION THREAD

**Discussion by yaronetokayer** [community/community#67650] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

I have several repositories on my local machine, but for only one of them, which is a python package, I get the following error message every few minutes:

fatal: git show-ref: bad ref refs/tags/Icon?
(0000000000000000000000000000000000000000)

I assume this has to do with the 'Icon' that mac OS automatically puts in every directory, but it is not an issue in any of my other repos.

I tried completely recreating the repository online and recloning it, in case it was an issue with the .gitignore file or something weird with how the OS was treating the directory, but the issue persists

<img width="400" alt="Screenshot 2023-09-19 at 7 27 53 PM" src="https://github.com/community/community/assets/109242322/7f3cf62f-fb6f-42e8-ae87-2b7ca7536fbf">

Furthermore, when I attempt to fetch the remote repo, I get the following message:
<img width="550" alt="Screenshot 2023-09-19 at 7 37 04 PM" src="https://github.com/community/community/assets/109242322/f51a3b06-fc1f-4be9-aff4-d041dec52a4d">

**Comment by seyLu** [comment, marked as answer]
#### Hi @yaronetokayer,

First delete the problematic references in git:

```bash
git update-ref -d <problematic_reference>
# eg. git update-ref -d refs/tags/Icon?
```

Then update `.gitignore`:

```bash
# General
.DS_Store
.AppleDouble
.LSOverride

# Icon must end with two \r
Icon

# Thumbnails
._*

# Files that might appear in the root of a volume
.DocumentRevisions-V100
.fseventsd
.Spotlight-V100
.TemporaryItems
.Trashes
.VolumeIcon.icns
.com.apple.timemachine.donotpresent

# Directories potentially created on remote AFP share
.AppleDB
.AppleDesktop
Network Trash Folder
Temporary Items
.apdisk
```

You can read more on:
- [Git docs: `git update-ref`](https://git-scm.com/docs/git-update-ref)
- [Gitignore template: macOs](https://github.com/github/gitignore/blob/main/Global/macOS.gitignore)

**Reply by yaronetokayer** [reply]
Thank you!  In order to fully resolve the issue, each of the 'Icon?'s in .git/refs and all of its subdirectories had to be deleted.  I did this manually instead of using `git update-ref`, since doing so returned `zsh: no matches found: refs/tags/Icon?`.
Once all of those were deleted, the updates to .gitignore seemed to do the trick.

**Reply by juliaross2** [reply]
@yaronetokayer how did you delete them manually? I am running into this same issue

**Reply by yaronetokayer** [reply]
@juliaross2 glad it was not just me!  In finder, you can view hidden files with command+shift+. and that will expose the 'Icon?' files, and then you can delete them from there like you would any other file.  Alternatively, in the terminal, 'ls -a' will show a list of all directory contents including the hidden files, and again you can delete them from there like you would any other file.
